packages/base/src/UI5Element.js

```diff
@@ -620,25 +622,12 @@ class UI5Element extends HTMLElement {
 	 * @private
 	 */
 	_updateShadowRoot() {
-		if (!this.constructor._needsShadowDOM()) {
-			return;
-		}
-
 		let styleToPrepend;
 		const renderResult = executeTemplate(this.constructor.template, this);
 
-		// IE11, Edge
-		if (window.ShadyDOM) {
-			createComponentStyleTag(this.constructor);
-		}
-
-		// Chrome
-		if (document.adoptedStyleSheets) {
+		if (document.adoptedStyleSheets) { // Chrome
 			this.shadowRoot.adoptedStyleSheets = getConstructableStyle(this.constructor);
-		}
-
-		// FF, Safari
-		if (!document.adoptedStyleSheets && !window.ShadyDOM) {
+		} else if (hasNativeSupport()) { // FF, Safari
```

Comment:
hasNativeSupport for what? I had to check the implementation, this name is too generic.